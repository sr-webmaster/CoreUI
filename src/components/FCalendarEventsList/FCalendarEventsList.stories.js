import { MAIN } from '../../../.storybook/categories'
import FCalendarEventsList from './FCalendarEventsList.vue'
import { action } from '@storybook/addon-actions'

const getTitle = () => {
  return `${MAIN}|FCalendarEventsList`
}

const events = [
  {
    name: 'Conference Call',
    date_time: '2019-01-01 11:28:00',
    type: {
      name: 'Phonecall'
    }
  },
  {
    name: 'Conference Call 1',
    date_time: '2019-02-28 14:30:00',
    type: {
      name: 'Phonecall'
    }
  },
  {
    name: 'Conference Call 2  with a very long title that may just go on forever and ever...  But seriously, why is this event title so long?  I do not really know, do you?',
    date_time: '2019-12-31 18:00:00',
    type: {
      name: 'Phonecall'
    }
  }
]

export default {
  title: getTitle(),
  id: 'FCalendarEventsList'
}

/**
 * https://storybook.js.org/docs/guides/guide-vue/#step-4-write-your-stories
 */
export const WithEvents = () => ({
  components: {
    FCalendarEventsList
  },
  data: () => ({
    events
  }),
  template: `
    <f-calendar-events-list :events="events" @edit="editActivity" @delete="deleteActivity" />
  `,
  methods: {
    editActivity (event) {
      action('edit')(event)
    },
    deleteActivity (event) {
      action('delete')(event)
    }
  }
})
