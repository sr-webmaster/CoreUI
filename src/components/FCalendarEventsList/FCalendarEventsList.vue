<template>
  <v-list>
    <f-calendar-event
      v-for="(event, idx) in events"
      :key="idx"
      :date="event.date_time"
      :title="event.name"
      :subtitle="event.type.name"
      class="f-calendar-events-list__item py-2 px-0"
    >
      <template v-slot:append>
        <v-menu>
          <template v-slot:activator="{ on }">
            <v-btn
              flat
              small
              class="f-calendar-events-list__item-menu-button ma-0 px-1"
              v-on="on"
            >
              Edit
              <v-icon
                right
                dark
                class="mx-0"
              >
                expand_more
              </v-icon>
            </v-btn>
          </template>

          <v-list>
            <v-list-tile
              v-bind="$attrs"
              @click="editItem(event)"
            >
              <v-list-tile-title>Edit</v-list-tile-title>
            </v-list-tile>
            <v-list-tile
              v-bind="$attrs"
              @click="deleteItem(event)"
            >
              <v-list-tile-title>Delete</v-list-tile-title>
            </v-list-tile>
          </v-list>
        </v-menu>
      </template>
    </f-calendar-event>
  </v-list>
</template>

<script>
import FCalendarEvent from '@freshinup/core-ui/src/components/FCalendarEvent'

export default {
  components: {
    FCalendarEvent
  },
  props: {
    events: {
      type: Array,
      default: () => []
    }
  },
  methods: {
    editItem (value) {
      this.$emit('edit', value)
    },
    deleteItem (value) {
      this.$emit('delete', value)
    }
  }
}
</script>

<style>
.f-calendar-events-list__item-menu-button {
  min-width: auto;
}
</style>
