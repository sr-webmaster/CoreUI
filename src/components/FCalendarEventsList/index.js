import Component from './FCalendarEventsList'
export default Component
