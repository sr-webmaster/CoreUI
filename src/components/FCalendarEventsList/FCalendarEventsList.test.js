import { mount, shallowMount } from '@vue/test-utils'
import createLocalVue from 'vue-cli-plugin-freshinup-ui/utils/testing/createLocalVue'
import Component from '@freshinup/core-ui/src/components/FCalendarEventsList'
import { WithEvents } from './FCalendarEventsList.stories'

describe('FCalendarEventsList', () => {
  describe('Methods', () => {
    // Using shallowMount test the methods
    const event = { name: 'test event' }
    const wrapper = shallowMount(Component)
    const vm = wrapper.vm
    vm.editItem(event)
    expect(wrapper.emitted('edit')).toBeTruthy()
    expect(wrapper.emitted('edit')).toEqual([
      [{ name: event.name }]
    ])
    vm.deleteItem(event)
    expect(wrapper.emitted('delete')).toBeTruthy()
    expect(wrapper.emitted('delete')).toEqual([
      [{ name: event.name }]
    ])
  })

  describe('Visuals', () => {
    test('defaults', () => {
      // Use mount and snapshot assertions
      const wrapper = mount(WithEvents(), {
        localVue: createLocalVue().vue
      })

      expect(wrapper.isVueInstance()).toBe(true)
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
