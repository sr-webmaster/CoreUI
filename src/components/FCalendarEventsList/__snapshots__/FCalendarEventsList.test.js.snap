// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FCalendarEventsList Visuals defaults 1`] = `
<div
  class="v-list theme--light"
  role="list"
>
  <div
    class="px-0 f-calendar-events-list__item py-2 px-0"
    role="listitem"
  >
    <div
      class="v-list__tile theme--light"
    >
      <div
        class="v-list__tile__action v-list__tile__action--stack white--text f-calendar-event__cal text-xs-center py-0 pr-2 mr-2 pl-0"
      >
        <div
          class="primary f-calendar-event__month-day"
        >
          <div
            class="f-calendar-event__month"
          >
            
          Jan
        
          </div>
           
          <div
            class="f-calendar-event__date headline"
          >
            
          1
        
          </div>
        </div>
         
        <div
          class="f-calendar-event__time secondary caption pa-1"
        >
          
        11:28 AM
      
        </div>
      </div>
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          <div
            class="f-calendar-event__title subtitle"
          >
            <span
              class="primary--text"
            >
              
            Conference Call
          
            </span>
          </div>
        </div>
         
        <div
          class="v-list__tile__sub-title"
        >
          <div
            class="f-calendar-event__subtitle body-2 grey--text"
          >
            
          Phonecall
        
          </div>
        </div>
         
        <div
          class="v-menu"
        >
          <button
            class="f-calendar-events-list__item-menu-button ma-0 px-1 v-btn v-btn--flat v-btn--small theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
            Edit
            
              <i
                aria-hidden="true"
                class="v-icon mx-0 v-icon--right material-icons theme--dark"
              >
                expand_more
              </i>
            </div>
          </button>
          <div
            class="v-menu__content theme--light "
            style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
          >
             
            <div
              class="v-list theme--light"
              role="list"
            >
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__title"
                  >
                    Edit
                  </div>
                </a>
              </div>
               
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__title"
                  >
                    Delete
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="px-0 f-calendar-events-list__item py-2 px-0"
    role="listitem"
  >
    <div
      class="v-list__tile theme--light"
    >
      <div
        class="v-list__tile__action v-list__tile__action--stack white--text f-calendar-event__cal text-xs-center py-0 pr-2 mr-2 pl-0"
      >
        <div
          class="primary f-calendar-event__month-day"
        >
          <div
            class="f-calendar-event__month"
          >
            
          Feb
        
          </div>
           
          <div
            class="f-calendar-event__date headline"
          >
            
          28
        
          </div>
        </div>
         
        <div
          class="f-calendar-event__time secondary caption pa-1"
        >
          
        2:30 PM
      
        </div>
      </div>
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          <div
            class="f-calendar-event__title subtitle"
          >
            <span
              class="primary--text"
            >
              
            Conference Call 1
          
            </span>
          </div>
        </div>
         
        <div
          class="v-list__tile__sub-title"
        >
          <div
            class="f-calendar-event__subtitle body-2 grey--text"
          >
            
          Phonecall
        
          </div>
        </div>
         
        <div
          class="v-menu"
        >
          <button
            class="f-calendar-events-list__item-menu-button ma-0 px-1 v-btn v-btn--flat v-btn--small theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
            Edit
            
              <i
                aria-hidden="true"
                class="v-icon mx-0 v-icon--right material-icons theme--dark"
              >
                expand_more
              </i>
            </div>
          </button>
          <div
            class="v-menu__content theme--light "
            style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
          >
             
            <div
              class="v-list theme--light"
              role="list"
            >
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__title"
                  >
                    Edit
                  </div>
                </a>
              </div>
               
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__title"
                  >
                    Delete
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="px-0 f-calendar-events-list__item py-2 px-0"
    role="listitem"
  >
    <div
      class="v-list__tile theme--light"
    >
      <div
        class="v-list__tile__action v-list__tile__action--stack white--text f-calendar-event__cal text-xs-center py-0 pr-2 mr-2 pl-0"
      >
        <div
          class="primary f-calendar-event__month-day"
        >
          <div
            class="f-calendar-event__month"
          >
            
          Dec
        
          </div>
           
          <div
            class="f-calendar-event__date headline"
          >
            
          31
        
          </div>
        </div>
         
        <div
          class="f-calendar-event__time secondary caption pa-1"
        >
          
        6:00 PM
      
        </div>
      </div>
       
      <div
        class="v-list__tile__content"
      >
        <div
          class="v-list__tile__title"
        >
          <div
            class="f-calendar-event__title subtitle"
          >
            <span
              class="primary--text"
            >
              
            Conference Call 2  with a very long title that may just go on forever and ever...  But seriously, why is this event title so long?  I do not really know, do you?
          
            </span>
          </div>
        </div>
         
        <div
          class="v-list__tile__sub-title"
        >
          <div
            class="f-calendar-event__subtitle body-2 grey--text"
          >
            
          Phonecall
        
          </div>
        </div>
         
        <div
          class="v-menu"
        >
          <button
            class="f-calendar-events-list__item-menu-button ma-0 px-1 v-btn v-btn--flat v-btn--small theme--light"
            type="button"
          >
            <div
              class="v-btn__content"
            >
              
            Edit
            
              <i
                aria-hidden="true"
                class="v-icon mx-0 v-icon--right material-icons theme--dark"
              >
                expand_more
              </i>
            </div>
          </button>
          <div
            class="v-menu__content theme--light "
            style="max-height: auto; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
          >
             
            <div
              class="v-list theme--light"
              role="list"
            >
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__title"
                  >
                    Edit
                  </div>
                </a>
              </div>
               
              <div
                role="listitem"
              >
                <a
                  class="v-list__tile v-list__tile--link theme--light"
                >
                  <div
                    class="v-list__tile__title"
                  >
                    Delete
                  </div>
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
